File: MJRefreshDemo/MJRefreshDemo/ViewController.m
//
//  ViewController.m
//  MJRefreshDemo
//
//  Created by 英策网络科技 on 16/5/7.
//  Copyright © 2016年 英策网络科技. All rights reserved.
//

#import "ViewController.h"
#import <MJRefresh.h>
#import <AFNetworking.h>
@interface ViewController ()<UITableViewDelegate,UITableViewDataSource>
{
    NSInteger page;
}
@property(nonatomic,strong)UITableView*mainTableView;
@property(nonatomic,strong)NSMutableArray*dataArray;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    page=1;
    self.dataArray=[NSMutableArray array];
    self.mainTableView=[[UITableView alloc]initWithFrame:CGRectMake(0, 300, self.view.frame.size.width, self.view.frame.size.height-300) style:UITableViewStylePlain];
    self.mainTableView.dataSource=self;
    self.mainTableView.delegate=self;
    self.mainTableView.tableFooterView=[[UIView alloc]initWithFrame:CGRectZero];
    [self.view addSubview:self.mainTableView];
    
    MJRefreshGifHeader*m_head=[[MJRefreshGifHeader alloc]init];
    [m_head setRefreshingTarget:self refreshingAction:@selector(refreshData)];
    self.mainTableView.mj_header=m_head;
    
    MJRefreshAutoFooter*m_foot=[[MJRefreshAutoFooter alloc]init];
    [m_foot setRefreshingTarget:self refreshingAction:@selector(addData)];
    [m_foot setAutomaticallyRefresh:YES];
    [m_foot setTriggerAutomaticallyRefreshPercent:0.8];
    self.mainTableView.mj_footer=m_foot;
    
    [self.mainTableView.mj_header beginRefreshing];
}
-(UITableViewCell*)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    static NSString*cellID=@"cellID";
    UITableViewCell*cell=[tableView dequeueReusableCellWithIdentifier:cellID];
    if (cell==nil) {
        cell=[[UITableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellID];
    }
    cell.textLabel.text=self.dataArray[indexPath.row];
    return cell;
}
-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return _dataArray.count;
}
-(void)addData{
    page++;
    [self requestHttp];
}
-(void)refreshData{
    page=1;
    [self.dataArray removeAllObjects];
    [self requestHttp];
}
-(void)requestHttp{
    NSDictionary*dic=[NSDictionary dictionaryWithObjectsAndKeys:@"73",@"uid",@"goodsids",@"show",[NSNumber numberWithInteger:page],@"page", nil];
    AFHTTPRequestOperationManager*manager=[AFHTTPRequestOperationManager manager];
    manager.responseSerializer.acceptableContentTypes=[NSSet setWithObject:@"text/html"];
    [manager GET:@"http://heng-s-micro-mall.hixiaoyuan.cn/addons/ewei_shop/core/api/shopmanage.php" parameters:dic success:^(AFHTTPRequestOperation * _Nonnull operation, id  _Nonnull responseObject) {
        NSDictionary*dataDic=[responseObject objectForKey:@"data"];
        NSArray*secondArr=dataDic[@"goods"];
        [secondArr enumerateObjectsUsingBlock:^(NSDictionary*  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
            [self.dataArray addObject:obj[@"title"]];
            [self.mainTableView reloadData];
            [self.mainTableView.mj_header endRefreshing];
            [self.mainTableView.mj_footer endRefreshing];
        }];
    } failure:^(AFHTTPRequestOperation * _Nullable operation, NSError * _Nonnull error) {
        
    }];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: swiftSeven/swiftSeven/ViewController.swift
//
//  ViewController.swift
//  swiftSeven
//
//  Created by 英策网络科技 on 16/8/4.
//  Copyright © 2016年 英策网络科技. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let city = ["1","2","3","4","5"]
        print(倒序("2", b: "1"))
        let city1 = city.sort(倒序)
        print(city1)
        let city2 = city.sort({(a:String,b:String) -> Bool in
        return a<b
        })
        print(city2)
        let city3 = city.sort{(a,b) in
            a>b
        }
        print(city3)
        let city4 = city.sort{
        $0 > $1
        }
        print(city4)
        let city5 = city.sort(){
           $0>$1
        }
        print(city5)
        let 闭包1 = 闭包(1, b: 2){(a:Int,b:Int) -> Int in
            a+b
        }
        print(闭包1)
        let 闭包2 = 闭包(1, b: 2, c: {a,b in
            a-b
        })
        print(闭包2)
    }
    func 倒序(a:String,b:String)->Bool{
    return a>b
    }
    func 闭包(a:Int,b:Int,c:(Int,Int)->Int)->Int{
    return c(a,b)
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

